src/ir_Haier.cpp

```diff
@@ -654,27 +654,101 @@ void IRHaierAC176::setMode(uint8_t mode) {
 /// @return The current operating mode setting.
 uint8_t IRHaierAC176::getMode(void) const { return _.Mode; }
 
-/// Set the temperature.
-/// @param[in] celsius The temperature in degrees celsius.
-void IRHaierAC176::setTemp(const uint8_t celsius) {
-  uint8_t temp = celsius;
-  if (temp < kHaierAcMinTemp)
-    temp = kHaierAcMinTemp;
-  else if (temp > kHaierAcMaxTemp)
-    temp = kHaierAcMaxTemp;
+/// Set the default temperature units to use.
+/// @param[in] on Use Fahrenheit as the units.
+///   true is Fahrenheit, false is Celsius.
+void IRHaierAC176::setUseFahrenheit(const bool on) { _.UseFahrenheit = on; }
 
-  uint8_t old_temp = getTemp();
-  if (old_temp == temp) return;
-  if (old_temp > temp)
-    _.Button = kHaierAcYrw02ButtonTempDown;
-  else
-    _.Button = kHaierAcYrw02ButtonTempUp;
-  _.Temp = temp - kHaierAcMinTemp;
+/// Get the default temperature units in use.
+/// @return true is Fahrenheit, false is Celsius.
+bool IRHaierAC176::getUseFahrenheit(void) const { return _.UseFahrenheit; }
+
+/// Set the temperature.
+/// @param[in] degree The temperature in degrees.
+/// @param[in] fahrenheit Use units of Fahrenheit and set that as units used.
+void IRHaierAC176::setTemp(const uint8_t degree, const bool fahrenheit) {
+  if (fahrenheit) {
+    uint8_t temp = degree;
+    if (temp < kHaierAcYrw02MinTempF)
+      temp = kHaierAcYrw02MinTempF;
+    else if (temp > kHaierAcYrw02MaxTempF)
+      temp = kHaierAcYrw02MaxTempF;
+
+    uint8_t old_temp = getTemp();
+    if (old_temp == temp) return;
+    if (old_temp > temp)
+      _.Button = kHaierAcYrw02ButtonTempDown;
+    else
+      _.Button = kHaierAcYrw02ButtonTempUp;
+
+    if (degree >= 77) { temp++; }
+    if (degree >= 79) { temp++; }
```

Comment:
You've got `78` & `79` in `getTemp()`. Something is probably wrong.

Reply:
Fix me if I wrong, but I see no error in this code. There we have a "strange" method of encoding degree Fahrenheit in Haier protocol. It use 4 bits in `Temp` and 1 extra bit in `ExtraDegreeF`, 5 bits in all.
Minimal temperature in Fahrenheit degree is 60F representet by `Temp` = `0b0000` and `ExtraDegreeF` = `0b0`, so 60F represented by 5 bits `0b00000`.
61F representation is `0b00001`
62F -> `0b00010`
63F -> `0b00011` // and so on...
***
76F -> `0b10000` // it is as expected
77F -> `0b10010` // in this case expected `0b10001`, but this bits combination not used
78F -> `0b10100` // and in this case expected `0b10011`, but this bits combination not used too
79F -> `0b10101` // now represented value as expeted, again
80F -> `0b10110`
***
86F -> `0b11100`
Combinations  `0b11101`,  `0b11110` and  `0b11111` not used.

So, using this code I try to skip unused bits combination `0b10001` and `0b10011`.

Reply:
My point was, you've got `77` & `79` in one place, and `78`, & `79` in another. If they are both correct, there should be a comment or something to explain that, otherwise someone might make the same conclusion I did.